spring-xd-runner/src/main/java/org/springframework/xd/dirt/plugins/job/JobPluginMetadataResolver.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.plugins.job;
+
+import static org.springframework.xd.module.ModuleType.job;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import javax.validation.constraints.AssertTrue;
+
+import org.springframework.xd.dirt.plugins.job.support.listener.XDJobListenerConstants;
+import org.springframework.xd.module.ModuleDefinition;
+import org.springframework.xd.module.options.ModuleOptionsMetadata;
+import org.springframework.xd.module.options.ModuleOptionsMetadataResolver;
+import org.springframework.xd.module.options.PojoModuleOptionsMetadata;
+import org.springframework.xd.module.options.spi.ModuleOption;
+import org.springframework.xd.module.options.spi.ProfileNamesProvider;
+import org.springframework.xd.rest.domain.util.TimeUtils;
+
+
+/**
+ * A {@link ModuleOptionsMetadataResolver} that will dynamically add the module options (such as {@code makeUnique})
+ * supported by job modules.
+ *
+ * @see JobPlugin
+ * @author Eric Bottard
+ * @author Ilayaperumal Gopinathan
+ */
+public class JobPluginMetadataResolver implements ModuleOptionsMetadataResolver {
```

Comment:
interested in what @mminella might have to add, but we could probably do without any job-related metadata support at this point